core/storage/email/gae_models.py

```diff
@@ -138,6 +175,15 @@ def get_by_hash(cls, email_hash, sent_datetime_lower_bound=None):
     def _generate_hash(cls, recipient_id, email_subject, email_body):
         """Generate hash for a given recipient_id, email_subject and cleaned
         email_body.
+
+        Args:
+            recipient_id: str. The user ID of the email recipient.
+            email_subject: str. The subject line of the email.
+            email_body: str. The HTML content of the email body.
+
+        Returns:
+            Generated hash value of the email in the messages list.
```

Comment:
str. The generated hash value of the given email.

(I don't understand the existing docstring. What messages list are you referring to?)